tests/framework/db/ColumnSchemaTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace yiiunit\framework\db;
+
+use yiiunit\framework\db\DatabaseTestCase;
+use yii\db\ColumnSchema;
+
+/**
+ * @group db
+ * @group mysql
+ */
+class ColumnSchemaTest extends DatabaseTestCase
+{
+
+    public function testDbTypecastNoCommas()
+    {
+        $locale = setlocale(LC_NUMERIC, 0);
+        if (false === $locale) {
+            $this->markTestSkipped('Your platform does not support locales.');
+        }
+
+        try {
+            // This one sets decimal mark to comma sign
+            setlocale(LC_NUMERIC, 'ru_RU.utf8');
+            $cschemaDouble = new ColumnSchema([
+                'type' => 'double',
+                'dbType' => 'double',
+                'phpType' => 'double',
+            ]);
+            $this->assertEquals('1.23', (string) $cschemaDouble->dbTypecast('1.23'));
+            $this->assertEquals('1.23', (string) $cschemaDouble->dbTypecast(1.23));
+            $this->assertEquals('2.0E+30', (string) $cschemaDouble->dbTypecast(2e+30));
```

Comment:
As far as I remember (and I do not quite remember it) persistence level itself (PDO, whatever) performs string cast. So even if you have tests passing at framework level, they would fail on functional tests with DB involved. I might had little confidence in writing functional tests in the time this PR was made.